platine-management-service/src/main/java/fr/insee/survey/datacollectionmanagement/questioning/service/impl/QuestioningEventServiceImpl.java

```diff
@@ -144,18 +157,25 @@ public void postExpertEvent(UUID id, ExpertEventDto expertEventDto) {
         candidate.setType(expertEventDto.type());
         candidate.setDate(new Date());
 
+        boolean saved = false;
```

Comment:
suggestion: use a method shouldSaveEvent? With this method, there is juste one block code and no saved boolean:
```
            candidate = questioningEventRepository.save(candidate);
            questioning.getQuestioningEvents().add(candidate);
            refreshHighestEvent(questioning.getId());

 shouldSaveEvent(Event candidate, Event lastEvent) {
    return (lastEvent == null &&
               (candidate.getType() == TypeQuestioningEvent.EXPERT
                || candidate.getType() == TypeQuestioningEvent.VALID))
           ||
           (lastEvent != null
               && candidate.getType() != lastEvent.getType()
               && candidate.getType() != TypeQuestioningEvent.EXPERT);
}

```